### ISSUE TITLE
Shellbags table returning "Invalid string position" with OneNote entries

### ISSUE THREAD

**Issue by allwalte** [osquery/osquery#8041]
# Bug report
When there is a specific type of entry in shellbags (ShellBags Explorer classifies these as "OneNote" entries, specifically "Value: OneNote
Shell Type: Root folder: GUID"), queries to the shellbags table return "Invalid string position".   The hex for these entries is ```14-00-1F-80-8F-00-F0-12-13-21-2A-48-80-B3-FD-5E-91-C1-23-13-00-00```.  Looking through the specifications [here](https://github.com/libyal/libfwsi/blob/main/documentation/Windows%20Shell%20Item%20format.asciidoc#3-type-indicator-based-shell-items) I can see that this is a CLSID_ShellDesktop class type.  Looking at the sort index entry of 80, that same spec indicates that this should be a "My Games" entry?  However, it seems like that's changed (and Shellbags Explorer is recognizing this new value correctly, but I have no idea where he's getting the info because I can't find documentation of 0x80 being used to indicate OneNote anywhere). 

The subkeys of this are in the form of a URL, specifically a personal sharepoint site.  This is where we run into the problem.  The subkey starts with ```be,00,00,00,00,b0,01,c0,b8,00,00,00,04,00,00,00,07,00,00,00,40,00,00,
  00,68,00,74,00,74,00,70,00,73,00,3a,00,2f,00,2f``` , which is the various type and size indicators, and then it starts to write the URL - "68,00,74,00,74,00,70,00,73,00,3a,00,2f,00,2f" or "https://".  However, since the parsing code thinks this is a path on the local drive, it's hitting [this code](https://github.com/osquery/osquery/blob/7b5c9ebd1a2dfd2c311669994729c97db3636938/osquery/tables/system/windows/shellbags.cpp#L72), seeing the "2F" slash at byte 80, and erroneously determining from that that this is a zip file.

It's then going into ZipContentItem and hitting an issue [here](https://github.com/osquery/osquery/blob/7b5c9ebd1a2dfd2c311669994729c97db3636938/osquery/utils/windows/shellitem.cpp#L239), since the URL is still continuing to be written out at this point so it's seeing content it doesn't expect.  I've confirmed that if I truncate the entry exactly at 176 (or shorter) then there's no error, it just returns ```[UNKNOWN VARIABLE FORMAT]``` for that entry and goes on its merry way.  Any longer and the error happens.  That holds true across each of the three machines with this problem, even though the URLs are different so the truncation happens at a different point in the URL - the change always takes effect with the truncation in that exact spot, regardless of what chars come before (or were chopped off).

**TL;dr - It seems like this 0x80 Sort Index has changed, and needs a different parsing case.**

Additional info:  The GUID the Shellbags Explorer creator is listing as OneNote, which corresponds to the data in this key, is [12f0008f-2113-482a-80b3-fd5e91c12313](https://github.com/EricZimmerman/GuidMapping/blob/master/Resources/GuidToName.txt#L11189).  Looking for this GUID in the reg I can see a key: ```Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Office\ClickToRun\REGISTRY\MACHINE\Software\Microsoft\Windows\CurrentVersion\Explorer\Desktop\Namespace\{12F0008F-2113-482A-80B3-FD5E91C12313}``` with the value "Web Sites", so the GUID does seem to check out but I can't see anything tying it into Sort Index 0x80.  Hopefully someone more familiar with all this than I can figure out a good source to confirm this data though?

### What operating system and version are you using?
Windows 10.0.22621

### What version of osquery are you using?
5.8.2

### What steps did you take to reproduce the issue?
Found multiple machines with the error, narrowed down reg keys until I determined that it was the same type of key in all cases.  Deleted the key and things worked normally in each case.

### What did you expect to see?
```select * from shellbags;``` returning data

### What did you see instead?
```select * from shellbags;``` returns "Invalid string position"
